src/Content/Backend/Solution/Monaco.Template.Backend.Common.Domain.Tests/ValueObjectTests.cs

```diff
@@ -11,46 +11,60 @@ namespace Monaco.Template.Backend.Common.Domain.Tests;
 public class ValueObjectTests
 {
 	[Theory(DisplayName = "New ValueObject instance succeeds")]
-	[AnonymousData]
+	[AutoDomainData]
 	public void NewValueObjectInstanceSucceeds(string field1, string? field2)
 	{
 		var sut = new DummyValueObject(field1, field2);
 
-		sut.Field1.Should().Be(field1);
-		sut.Field2.Should().Be(field2);
+		sut.Field1
+		   .Should()
+		   .Be(field1);
+		sut.Field2
+		   .Should()
+		   .Be(field2);
 	}
 
 	[Theory(DisplayName = "Different ValueObject instances with same values are equal")]
-	[AnonymousData]
+	[AutoDomainData]
 	public void DifferentValueObjectInstancesWithSameValuesAreEqual(string field1, string? field2)
 	{
 		var val1 = new DummyValueObject(field1, field2);
 		var val2 = new DummyValueObject(field1, field2);
 
-		(val1 == val2).Should().BeTrue();
-		val1.Equals(val2).Should().BeTrue();
+		(val1 == val2).Should()
+					  .BeTrue();
+		val1.Equals(val2)
+			.Should()
+			.BeTrue();
 	}
 
 	[Theory(DisplayName = "Different ValueObject instances with same values are not equal")]
-	[AnonymousData]
+	[AutoDomainData]
 	public void DifferentValueObjectInstancesWithDifferentValuesAreNotEqual(string field1, string? field2, string? field3)
 	{
 		var val1 = new DummyValueObject(field1, field2);
 		var val2 = new DummyValueObject(field1, field3);
 
-		(val1 != val2).Should().BeTrue();
-		val1.Equals(val2).Should().BeFalse();
+		(val1 != val2).Should()
+					  .BeTrue();
+		val1.Equals(val2)
+			.Should()
+			.BeFalse();
 	}
 
 	[Theory(DisplayName = "Different ValueObject instances with same values are not equal")]
-	[AnonymousData]
+	[AutoDomainData]
 	public void ValueObjectComparedAgainstNullIsNotEqual(string field1, string? field2)
 	{
-		var sut = new DummyValueObject(field1, field2);
```

Comment:
Not sure I understand the purpose of this change?

Reply:
If I do it the previous way it will give a warning regarding the null comparison being false always. So I made it nullable and made the initialization happen in a delegate so it's not so straightforward to analyze by the IDE and not get a permanent warning about whether it is always false or not.
Do you think we could refactor it some other way?

Reply:
I'm not sure we need the test at all? Maybe I'm missing the point but why would creating any object and assigning it to a variable result in that variable being null?

Reply:
These were tests destined to cover different scenarios of Equals method and == operator for Value Objects. ValueObject is still a class and not a record, so there are overrides in place to properly mimic Records behavior.

Reply:
Ah yes of course, makes sense now, wasn't thinking of the operator overloading!

I guess it's fine overall then, it just seemed a convoluted refactor to avoid a warning. But if it works and serves the intended purpose then that's cool, and it's not a test/code that needs to be reviewed or understood regularly.

Reply:
Yeah, it doesn't really make me happy, but I wasn't sure having the warning disabled with a comment was much better... So I decided to refactor it like the initialization of the variable was deferred and then compared...
Feel free to chime in with a possible alternative :)